app/src/main/java/com/shagalalab/constitution/ui/base/SearchableFragment.kt

```diff
@@ -0,0 +1,49 @@
+package com.shagalalab.constitution.ui.base
+
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.widget.SearchView
+import androidx.core.view.MenuItemCompat
+import androidx.fragment.app.Fragment
+import com.shagalalab.constitution.R
+
+open class SearchableFragment(resId: Int) : Fragment(resId) {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        setHasOptionsMenu(true)
+        super.onCreate(savedInstanceState)
+    }
+
+    private lateinit var submitText: (text: String?) -> Unit
+
+    fun setSubmitText(submitText: (text: String?) -> Unit) {
+        this.submitText = submitText
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.menu, menu)
+        val searchItem: MenuItem = menu.findItem(R.id.menu)
+        val searchView: SearchView =
+            MenuItemCompat.getActionView(searchItem) as SearchView
+        val id = searchView.context.resources.getIdentifier(
+            "androidx.appcompat.R.id.search_src_text",
+            null,
+            null
+        )
+        searchView.setOnQueryTextListener(object :
+            SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(p0: String?): Boolean {
+                submitText.invoke(p0)
+                return false
```

Comment:
If you consumed the event, then you need to return `true`, it is like saying, "all good, I have taken care of this event myself". If you return `false`, it will say to its parent: "Oh, I cannot handle this event, so please do it yourself".